src/Controller/EmbargoesIpAccessDeniedController.php

```diff
@@ -3,42 +3,81 @@
 namespace Drupal\embargoes\Controller;
 
 use Drupal\Core\Controller\ControllerBase;
-use Drupal\Core\Render\Markup;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpFoundation\Request;
 
 /**
  * Class EmbargoesLogController.
  */
 class EmbargoesIpAccessDeniedController extends ControllerBase {
 
-  public function response() {
+  /**
+   * The HTTP request.
+   *
+   * @var Symfony\Component\HttpFoundation\Request
+   */
+  protected $request;
+
+  /**
+   * Constructs an IP access denied controller.
+   *
+   * @param Symfony\Component\HttpFoundation\Request $request
+   *   The current request.
+   */
+  public function __construct(Request $request = NULL) {
+    $this->request = $request;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('request_stack')->getCurrentRequest());
+  }
 
-    $allowed_ranges = explode('.', $_GET['ranges']);
-    $host = \Drupal::request()->getSchemeAndHttpHost();
-    $path = $_GET['path'];
-    $requested_resource = $host . $path;
-    $contact_email = \Drupal::config('embargoes.settings')->get('embargo_contact_email');
+  /**
+   * Helper function to attempt to get the current request.
+   *
+   * @return string|null
+   *   The requested resource, or NULL if there is no current request.
+   */
+  protected function getRequestedResource() {
+    if (!is_null($this->request)) {
+      $path = $this->request->query->get('path');
+      $host = $this->request->getSchemeAndHttpHost();
+      return "{$host}{$path}";
+    }
+  }
 
-    $message = "<p>Your request for the following resource could not be resolved:<br/><strong>{$requested_resource}</strong></p><br/>";
-    $message .= "<p>Access to this resource is restricted to the following networks:<br/><ul>";
-    foreach ($allowed_ranges as $allowed_range) {
-      $allowed_range_entity = \Drupal::entityTypeManager()->getStorage('embargoes_ip_range_entity')->load($allowed_range);
+  /**
+   * Formats a response for an IP access denied page.
+   *
+   * @return array
+   *   Renderable array of markup for IP access denied.
+   */
+  public function response() {
+    $requested_resource = $this->getRequestedResource();
+    $contact_email = $this->config('embargoes.settings')->get('embargo_contact_email');
+    $ranges = [];
+    foreach ($this->request->query->get('ranges') as $allowed_range) {
```

Comment:
Getting this:
`Warning: Invalid argument supplied for foreach() in Drupal\embargoes\Controller\EmbargoesIpAccessDeniedController->response() (line 63 of modules/contrib/embargoes/src/Controller/EmbargoesIpAccessDeniedController.php). `
...should probably check that getting 'ranges' returns an array before putting it in a foreach.